pkg/storaged/details.js

```diff
@@ -269,15 +269,24 @@
                                       options["config-items"] = { t: 'a(sa{sv})', v: config_items };
 
                                   if (create_partition) {
-                                      if (vals.type == "dos-extended")
+                                      if (vals.type == "dos-extended") {
                                           return block_ptable.CreatePartition(start, vals.size, "0x05", "", { });
-                                      else if (vals.type == "empty")
+                                      } else if (vals.type == "empty") {
                                           return block_ptable.CreatePartition(start, vals.size, "", "", { });
-                                      else
+                                      } else if (block_ptable.CreatePartitionAndFormat) {
                                           return block_ptable.CreatePartitionAndFormat (start, vals.size, "", "", { },
                                                                                         vals.type, options);
-                                  } else
+                                      } else {
+                                          return block_ptable.CreatePartition(start, vals.size, "", "", { })
+                                              .then(function(partition) {
+                                                  return block.Format(vals.type, options).then(function() {
```

Comment:
This should call format on client.blocks[partition], not on block which refers to the whole disk at this point.